Vagrantfile

```diff
@@ -680,91 +644,27 @@ Vagrant.configure('2') do |config|
   # Provisioning
   #
   # Process one or more provisioning scripts depending on the existence of custom files.
-  #
-  # provison-pre.sh acts as a pre-hook to our default provisioning script. Anything that
-  # should run before the shell commands laid out in provision.sh (or your provision-custom.sh
-  # file) should go in this script. If it does not exist, no extra provisioning will run.
-  if File.exist?(File.join(vagrant_dir, 'provision', 'provision-pre.sh'))
-    config.vm.provision 'pre', type: 'shell', keep_color: true, path: File.join('provision', 'provision-pre.sh')
-  end
 
-  # provision.sh or provision-custom.sh
-  #
-  # By default, Vagrantfile is set to use the provision.sh bash script located in the
-  # provision directory. If it is detected that a provision-custom.sh script has been
-  # created, that is run as a replacement. This is an opportunity to replace the entirety
-  # of the provisioning provided by default.
-  if File.exist?(File.join(vagrant_dir, 'provision', 'provision-custom.sh'))
-    config.vm.provision 'custom', type: 'shell', keep_color: true, path: File.join('provision', 'provision-custom.sh')
-  else
-    config.vm.provision 'default', type: 'shell', keep_color: true, path: File.join('provision', 'provision.sh')
-  end
-
-  # Provision the dashboard that appears when you visit vvv.test
-  config.vm.provision 'dashboard',
-                      type: 'shell',
-                      keep_color: true,
-                      path: File.join('provision', 'provision-dashboard.sh'),
-                      args: [
-                        vvv_config['dashboard']['repo'],
-                        vvv_config['dashboard']['branch']
-                      ]
-
-  vvv_config['utility-sources'].each do |name, args|
-    config.vm.provision "utility-source-#{name}",
-                        type: 'shell',
-                        keep_color: true,
-                        path: File.join('provision', 'provision-utility-source.sh'),
-                        args: [
-                          name,
-                          args['repo'].to_s,
-                          args['branch']
-                        ]
+  ARGV.each do |arg|
+    match = arg.match(/(--provision-with=)(.*)/i);
+    if match
+      _prefix, req_prov = match.captures
+      config.vm.provision "#{req_prov}", type: 'shell', keep_color: true, path: File.join('provision', 'run-provisioners.sh'), args: [ req_prov ]
+    end
```

Comment:
can't we just keep our provisioners defined? It feels weird to do things like this, lets just have standard vagrant provisioners. Otherwise, we're going to need to implement white lists to avoid people doing silly things like `vagrant provison --provision-with="bananas in pyjamas are walking down the stairs"`.

Also it's really handy when you reprovision and it outright tells you what the provisioner names are so you can copy paste them into `--provision-with=`

Reply:
--provision-with already fails if you run some random provisioner that doesn't exist. In this case, it would just run nothing.

The idea of this PR is to detach as much as possible from vagrant, and just have it as a tool to create the VM and handle hosts files and shared folders. 

I've moved logic to a single place (the VM and bash) and removed the old ruby logic, if we keep the logic to run provisioners in ruby, and in bash, we have unnecessary code duplication. We have simple workarounds (just like this one) to keep backwards compatibility.